packages/snfoundry/scripts-ts/deploy.ts

```diff
@@ -1,47 +1,30 @@
+import yargs from "yargs";
 import {
   deployContract,
   executeDeployCalls,
   exportDeployments,
   deployer,
+  loadExistingDeployments,
 } from "./deploy-contract";
-import { green } from "./helpers/colorize-log";
+import { green, yellow } from "./helpers/colorize-log";
+
+const argv = yargs(process.argv.slice(2))
+  .options({
+    network: { type: "string", demandOption: true },
+    reset: { type: "boolean", default: false },
+  })
+  .parseSync();
 
-/**
- * Deploy a contract using the specified parameters.
- *
- * @example (deploy contract with contructorArgs)
- * const deployScript = async (): Promise<void> => {
- *   await deployContract(
- *     {
- *       contract: "YourContract",
- *       contractName: "YourContractExportName",
- *       constructorArgs: {
- *         owner: deployer.address,
- *       },
- *       options: {
- *         maxFee: BigInt(1000000000000)
- *       }
- *     }
- *   );
- * };
- *
- * @example (deploy contract without contructorArgs)
- * const deployScript = async (): Promise<void> => {
- *   await deployContract(
- *     {
- *       contract: "YourContract",
- *       contractName: "YourContractExportName",
- *       options: {
- *         maxFee: BigInt(1000000000000)
- *       }
- *     }
- *   );
- * };
- *
- *
- * @returns {Promise<void>}
- */
 const deployScript = async (): Promise<void> => {
+  if (argv.reset) {
```

Comment:
can we leave the `deployScript` as before, and use this args inside `deploy-contract.ts` only ?

Reply:
Yes we can... Sending a PR now

Reply:
i still see the `deploy.ts` file same as before, we should be able to put all the logic inside `deploy-contract.ts` @josephchimebuka

Reply:
I made some new updates and put the logic into deploy- contract.ts files

Reply:
Hello @josephchimebuka , thanks for this. The file 'deploy.ts' should remain as the main branch. ( including the comments ) 

All the logic goes into 'deploy-contract.ts'

Reply:
@0xquantum3labs I've returned the deploy.ts to the default one